src/Lightning.ts

```diff
@@ -315,8 +315,7 @@ export const LightningMixin = (superclass) => class extends superclass {
         }
 
         if (String(payeeUser._id) === String(this.user._id)) {
-          const error = 'User tried to pay themselves'
-          throw new SelfPaymentError(error, {logger: lightningLoggerOnUs})
+          throw new SelfPaymentError(undefined, {logger: lightningLoggerOnUs})
```

Comment:
if you never override the message field, why have a parameter for it ? 

Seems like it will be cleaner to remove this parameter until you need it

Reply:
We don't override it for this particular error type, but we do for other error types. We're trying to keep the error classes consistent, which is why this parameter hasn't been removed here (I did consider it).

Reply:
Oh, I see where your head is.

I would ask: what is the value of keeping error classes consistent? 

Each Error Class is fundamentally a different object. 

Trying to add this implicit connection between your classes leads to confusion, because people like me have no context that you are doing something like that.

If you want to connect these classes, would recommend using composition.

Reply:
> Oh, I see where your head is.
> 
> I would ask: what is the value of keeping error classes consistent?
> 
> Each Error Class is fundamentally a different object.
> 
> Trying to add this implicit connection between your classes leads to confusion, because people like me have no context that you are doing something like that.

I disagree. The reason we're trying to keep these classes consistent in their naming as well as parameters is because it removes a lot of guessing work as to how an error class in our codebase works. All of them accept the exact same arguments and behave more or less the same way.

I agree there's scope for improvement here (especially with the degree of repetition across the error classes). 

You're welcome to take a stab at improving the error handling if you'd like. We appreciate and welcome all contributions :)